datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -503,8 +503,9 @@ fn read_partition(
                     let proj_batch = partition_column_projector
                         .project(adapted_batch, &partitioned_file.partition_values);
 
-                    send_result(&response_tx, proj_batch)?;
-                    if limit.map(|l| total_rows >= l).unwrap_or(false) {
+                    let send_rt = send_result(&response_tx, proj_batch);
```

Comment:
When both the where statement and the limit statement are used, the value of the limit is None. As a result, the normal limit statement also triggers the channel close error.

Reply:
I think it makes sense not to error if `send_result` fails due to the receiver being closed. I checked that this is the only error `send_result` can make: https://github.com/apache/arrow-datafusion/blob/41b4e491663029f653e491b110d0b5e74d08a0b6/datafusion/core/src/physical_plan/file_format/parquet.rs#L309

👍